testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueries.java

```diff
@@ -108,28 +104,16 @@ public void testAggregationOverUnknown()
     }
 
     @Test
-    public void testLimitIntMax()
+    public void testLimitMax()
     {
+        // max int
         assertQuery("SELECT orderkey FROM orders LIMIT " + Integer.MAX_VALUE);
         assertQuery("SELECT orderkey FROM orders ORDER BY orderkey LIMIT " + Integer.MAX_VALUE);
-    }
 
-    @Test
-    public void testNonDeterministic()
-    {
-        MaterializedResult materializedResult = computeActual("SELECT rand() FROM orders LIMIT 10");
-        long distinctCount = materializedResult.getMaterializedRows().stream()
-                .map(row -> row.getField(0))
-                .distinct()
-                .count();
-        assertTrue(distinctCount >= 8, "rand() must produce different rows");
-
-        materializedResult = computeActual("SELECT apply(1, x -> x + rand()) FROM orders LIMIT 10");
-        distinctCount = materializedResult.getMaterializedRows().stream()
-                .map(row -> row.getField(0))
-                .distinct()
-                .count();
-        assertTrue(distinctCount >= 8, "rand() must produce different rows");
+        // max long; a connector may attempt a pushdown while remote system may not accept such high limit values
+        assertQuery("SELECT nationkey FROM nation LIMIT " + Long.MAX_VALUE, "SELECT nationkey FROM nation");
```

Comment:
Does this mean connectors need to reject LIMIT pushdown for values of limit higher than what they support?

Reply:
Correct. For example H2 accepts 32-bit (31-bit) limit values, and  fails for max long.

Reply:
Oh, and it fails for Phoenix too

Reply:
i am removing this test from this PR

Reply:
Extracted into https://github.com/trinodb/trino/pull/7169